lib/sentry/opentelemetry/span_processor.ex

```diff
@@ -0,0 +1,199 @@
+              Logger.error("Failed to send transaction to Sentry: #{inspect(error)}")
+              {:error, :invalid_span}
+          end
+        else
+          true
+        end
+
+      :ok = SpanStorage.remove_span(span_record.span_id)
+
+      result
+    else
+      true
+    end
+  end
+
+  @impl true
+  def force_flush(_config) do
+    :ok
+  end
+
+  # TODO: "Elixir.Oban.Stager process" is an internal span and it produces a lot of noise
+  #       so it makes sense to ignore it - should this be configurable?
+  defp build_transaction(%{name: "Elixir.Oban.Stager process"}, _), do: nil
+
+  defp build_transaction(root_span_record, child_span_records) do
+    root_span = build_span(root_span_record)
+    child_spans = Enum.map(child_span_records, &build_span(&1))
+
+    Transaction.new(%{
+      span_id: root_span.span_id,
+      transaction: transaction_name(root_span_record),
+      transaction_info: %{source: :custom},
+      contexts: %{
+        trace: build_trace_context(root_span_record),
+        otel: build_otel_context(root_span_record)
+      },
+      spans: [root_span | child_spans]
+    })
+  end
+
+  defp transaction_name(
+         %{attributes: %{unquote(to_string(MessagingAttributes.messaging_system())) => :oban}} =
+           span_record
+       ) do
+    span_record.attributes["oban.job.worker"]
+  end
+
+  defp transaction_name(span_record), do: span_record.name
+
+  defp build_trace_context(span_record) do
+    {op, description} = get_op_description(span_record)
+
+    %{
+      trace_id: span_record.trace_id,
+      span_id: span_record.span_id,
+      parent_span_id: span_record.parent_span_id,
+      op: op,
+      description: description,
+      origin: span_record.origin,
+      data: span_record.attributes
+    }
+  end
+
+  defp build_otel_context(span_record), do: span_record.attributes
+
+  defp get_op_description(
+         %{
+           attributes: %{
+             unquote(to_string(HTTPAttributes.http_request_method())) => http_request_method
+           }
+         } = span_record
+       ) do
+    op = "http.#{span_record.kind}"
+    client_address = Map.get(span_record.attributes, to_string(ClientAttributes.client_address()))
+    url_path = Map.get(span_record.attributes, to_string(URLAttributes.url_path()))
+
+    description =
+      to_string(http_request_method) <>
+        ((client_address && " from #{client_address}") || "") <>
+        ((url_path && " #{url_path}") || "")
+
+    {op, description}
+  end
+
+  defp get_op_description(
+         %{attributes: %{unquote(to_string(DBAttributes.db_system())) => _db_system}} =
+           span_record
+       ) do
+    db_query_text = Map.get(span_record.attributes, "db.statement")
+
+    {"db", db_query_text}
+  end
+
+  defp get_op_description(%{
+         attributes:
+           %{unquote(to_string(MessagingAttributes.messaging_system())) => :oban} = attributes
+       }) do
+    {"queue.process", attributes["oban.job.worker"]}
+  end
```

Comment:
I think we should aim to make this agnostic instead of oban specific, but does require upstream to be fixed with https://github.com/open-telemetry/opentelemetry-erlang-contrib/pull/436 so we can `messaging.operation.type` and `messaging.destination.name` for this.